doc/source/data/batch_inference.rst

```diff
@@ -0,0 +1,752 @@
+To run inference workloads effectively at scale, you need to
+
+- manage your compute infrastructure and cloud clusters
+- parallelize data processing and utilize all your cluster resources (CPUs & GPUs)
+- efficiently transferring data between cloud storage, CPUs for preprocessing, and GPUs for model inference.
+
+Here's a more realistic view of batch inference for modern AI applications:
+
+.. figure:: images/batch_inference_overview.png
+
+  Evaluating a batch of input data with a model to get predictions.
+
+Why use Ray for batch inference?
+---------------------------------
+
+There are many good reasons to use Ray for batch inference today, even if your current
+use case does not require scaling yet:
+
+1. **Faster and Cheaper for modern Deep Learning Applications**: Ray is built for
+    complex workloads and supports loading and preprocessing data with CPUs and model inference on GPUs.
+2. **Cloud, framework, and data format agnostic**. Ray Data works on any cloud provider,
+    any ML framework (like PyTorch and Tensorflow) and does not require a particular file format.
+3. **Out of the box scaling**: The same code that works on one machine also runs on a
+    large cluster without any changes.
+4. **Python first** You can express your inference job directly in Python instead of
+    YAML files or other formats.
+
+.. _batch_inference_quickstart:
+
+Quick Start
+-----------
+
+To get started with batch inference with Ray, the first thing you need to do is
+install Ray together with its data processing library Ray Data:
+
+.. code-block:: bash
+
+    pip install ray[data]
+
+Running batch inference is conceptually easy and requires just three simple steps
+that we'll walk through in this guide:
+
+1. Load your data into a Ray ``Datastream`` and optionally apply any preprocessing you need.
+2. Define your model for inference.
+3. Run inference on your data by using the ``.map_batches()`` method from Ray Data.
+
+The last step also defines how your batch processing job gets distributed across your (local) cluster.
+We start with very simple use cases here and build up to more complex ones in other guides and tutorials.
+
+.. note::
+
+    All advanced use cases ultimately boil down to extensions of the above three steps,
+    like loading and storing data from cloud storage, using complex preprocessing functions,
+    demanding model setups, additional postprocessing, or other customizations.
+    We'll cover these advanced use cases in the next sections.
+
+1. Loading and Preprocessing Data
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For this quick-start guide we'll simply use very small, in-memory data sets by
+leveraging common Python libraries like NumPy and Pandas.
+In general, once you loaded your datasets using Ray Data, you also want to apply some preprocessing steps.
+We skip this step here for simplicity.
+In any case, the result of this step is a Ray Datastream ``ds`` that we can use to run inference on.
+
+.. margin::
```

Comment:
Recommend moving this to the intro of the advanced section.

Reply:
but I need to tell users now that we do it later. it's a note for the impatient